### ISSUE TITLE
[BUG] Segmentation fault in v0.84.0

### ISSUE THREAD

**Issue by gnumoksha** [DataDog/dd-trace-php#1933]
### Bug description
Segmentation fault after execute symfony's bin/console:

```console
$ DATADOG_VERSION=0.84.0
$ cd /tmp && curl --silent --show-error --location --output datadog.deb "https://github.com/DataDog/dd-trace-php/releases/download/${DATADOG_VERSION}/datadog-php-tracer_${DATADOG_VERSION}_amd64.deb" && dpkg -i datadog.deb && rm -f datadog.deb

$ php bin/console
Segmentation fault (core dumped)
```

### PHP version
```console
$ php -v
PHP 8.1.2 (cli) (built: Jan 26 2022 16:40:42) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.2, Copyright (c) Zend Technologies
    with Xdebug v3.1.3, Copyright (c) 2002-2022, by Derick Rethans
    with ddtrace v0.84.0, Copyright Datadog, by Datadog
```

### Tracer version
```console
$ php -r "echo phpversion('ddtrace').PHP_EOL;"
0.84.0
```

### Installed extensions
```console
$ php -m                                                                                                                         
[PHP Modules]                                                                                                                                                  
amqp                                                                                                                                                           
apcu                                                                                                                                                           
bcmath                                                                                                                                                         
Core                                                                                                                                                           
ctype                                                                                                                                                          
curl                                                                                                                                                           
date                                                                                                                                                           
ddtrace                                                                                                                                                        
dom                                                                                                                                                            
fileinfo                                                                                                                                                       
filter                                                                                                                                                         
ftp                                                                                                                                                            
hash
iconv
intl
json
libxml
mbstring
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_sqlite
Phar
posix
readline
redis
Reflection
session
shmop
SimpleXML
sodium
SPL
sqlite3
standard
sysvmsg
sysvsem
sysvshm
tokenizer
xdebug
xml
xmlreader
xmlwriter
zip
zlib

[Zend Modules]
Xdebug
ddtrace
```

### OS info
```console
$ cat /etc/os-release | grep -E "(NAME)|(VERSION)"
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
```

**Comment by morrisonlevi** [comment]
Thanks for the report. What Symfony version is this? The output says, "Segmentation fault (core dumped)." Can you open this core, which is usually something like `gdb php core`, and run `backtrace`? The output is often helpful for these issues.

-----

I tried to reproduce by using PHP 8.1.16 (obtained from dockerhub). I installed composer from getcomposer.org, and then ran:

```
composer create-project symfony/skeleton:"6.2.*" my_project_directory
cd my_project_directory
```

And then your commands. However, I am on a Mac M1 so I used arm64 instead of amd64. No errors.

-----

Recently there has been a string of crashes reported to us that stem from issues in PHP itself, and upgrading the PHP version makes them go away. Given this, I recommend upgrading to the latest 8.1 version.

I'll make sure the tracer engineers see this issue, of course.

**Comment by gnumoksha** [comment]
The error is gone. I've upgraded the PHP version to 8.2.3 and changed the agent installation method to the following:
```console
$ curl -LO https://github.com/DataDog/dd-trace-php/releases/latest/download/datadog-setup.php \
    && php datadog-setup.php --php-bin=all \
    && rm -f datadog-setup.php
```

Also, this the backtrace (of other project that I didn't upgrade yet, also running symfony 6.2):

```console
root@8787ee757901:/var/www/app# gdb php core                                                                                                                   
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git                                                                                                                 
Copyright (C) 2021 Free Software Foundation, Inc.                                                                                                              
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>                                                                                  
This is free software: you are free to change and redistribute it.                                                                                             
There is NO WARRANTY, to the extent permitted by law.                                                                                                          
Type "show copying" and "show warranty" for details.                                                                                                           
This GDB was configured as "x86_64-linux-gnu".                                                                                                                 
Type "show configuration" for configuration details.                                                                                                           
For bug reporting instructions, please see:                                                                                                                    
<https://www.gnu.org/software/gdb/bugs/>.                                                                                                                      
Find the GDB manual and other documentation resources online at:                                                                                               
    <http://www.gnu.org/software/gdb/documentation/>.                                                                                                          
                                                                                                                                                               
For help, type "help".                                                                                                                                         
Type "apropos word" to search for commands related to "word"...                                                                                                
Reading symbols from php...                                                                                                                                    
(No debugging symbols found in php)                                                                                                                            
[New LWP 82]                                                                                                                                                   
[Thread debugging using libthread_db enabled]                                                                                                                  
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".                                                                                     
Core was generated by `php bin/console'.                                                                                                                       
Program terminated with signal SIGSEGV, Segmentation fault.                                                                                                    
#0  0x000055856b7bc79b in zend_observer_fcall_end ()
```
```backtrace
(gdb) backtrace
#0  0x000055856b7bc79b in zend_observer_fcall_end ()                                                                                                           
#1  0x000055856b78c167 in execute_ex ()                                                                                                                        
#2  0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c185d0) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#3  0x000055856b71a127 in zend_call_function ()                                                                                                                
#4  0x000055856b71a545 in zend_call_known_function ()                                                                                                          
#5  0x000055856b5ea67e in ?? ()                                                                                                                                
#6  0x00007f3f76bbc99d in xdebug_execute_internal (current_execute_data=0x7f3f76c18580, return_value=0x7f3f76c18400)                                           
    at /tmp/pear/temp/xdebug/src/base/base.c:897                                                                                                               
#7  0x00007f3f7693ffe6 in zai_interceptor_execute_internal_impl (execute_data=0x7f3f76c18580, return_value=0x7f3f76c18400, prev=<optimized out>)               
    at /home/circleci/datadog/tmp/build_extension/zend_abstract_interface/interceptor/php8/interceptor.c:694                                                   
#8  0x000055856b439a55 in ?? ()                                                                                                                                
#9  0x000055856b43a207 in ?? ()                                                                                                                                
#10 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c180d0) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#11 0x000055856b4399c1 in ?? ()                                                                                                                                
#12 0x000055856b43a207 in ?? ()                                                                                                                                
#13 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c18000) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#14 0x000055856b71a127 in zend_call_function ()                                                                                                                
#15 0x000055856b71a545 in zend_call_known_function ()                                                                                                          
#16 0x000055856b7afe04 in zend_objects_destroy_object ()                                                                                                       
#17 0x000055856b7b488d in zend_objects_store_del ()                                                                                                            
#18 0x000055856b439ae5 in ?? ()                                                                                                                                
#19 0x000055856b43a207 in ?? ()
#20 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c17d60) at /tmp/pear/temp/xdebug/src/base/base.c:779                                 [36/384]
#21 0x000055856b4399c1 in ?? ()                                                                                                                                
#22 0x000055856b43a207 in ?? ()                                                                                                                                
#23 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c17910) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#24 0x000055856b4399c1 in ?? ()                                                                                                                                
#25 0x000055856b43a207 in ?? ()                                                                                                                                
#26 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c17670) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#27 0x000055856b4399c1 in ?? ()                                                                                                                                
#28 0x000055856b43a207 in ?? ()                                                                                                                                
#29 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c17310) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#30 0x000055856b4399c1 in ?? ()                                                                                                                                
#31 0x000055856b43a207 in ?? ()                                                                                                                                
#32 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c16f20) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#33 0x000055856b4399c1 in ?? ()                                                                                                                                
#34 0x000055856b43a207 in ?? ()                                                                                                                                
#35 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c16c70) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#36 0x000055856b4399c1 in ?? ()                                                                                                                                
#37 0x000055856b43a207 in ?? ()                                                                                                                                
#38 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c16b80) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
--Type <RET> for more, q to quit, c to continue without paging--                                                                                               
#39 0x000055856b4399c1 in ?? ()                                                                                                                                
#40 0x000055856b43a207 in ?? ()                                                                                                                                
#41 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c169b0) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#42 0x000055856b4399c1 in ?? ()                                                                                                                                
#43 0x000055856b43a207 in ?? ()                                                                                                                                
#44 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c163e0) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#45 0x000055856b4399c1 in ?? ()                                                                                                                                
#46 0x000055856b43a207 in ?? ()                                                                                                                                
#47 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c16340) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#48 0x000055856b4399c1 in ?? ()                                                                                                                                
#49 0x000055856b43a207 in ?? ()                                                                                                                                
#50 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c16250) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#51 0x000055856b4399c1 in ?? ()                                                                                                                                
#52 0x000055856b43a207 in ?? ()                                                                                                                                
#53 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c161d0) at /tmp/pear/temp/xdebug/src/base/base.c:779                                         
#54 0x000055856b4399c1 in ?? ()
#55 0x000055856b43a207 in ?? ()
#56 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c15f20) at /tmp/pear/temp/xdebug/src/base/base.c:779
#57 0x000055856b4399c1 in ?? ()
#58 0x000055856b43a207 in ?? ()
#59 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c15370) at /tmp/pear/temp/xdebug/src/base/base.c:779
#60 0x000055856b4399c1 in ?? ()
#61 0x000055856b43a207 in ?? ()
#62 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c150c0) at /tmp/pear/temp/xdebug/src/base/base.c:779
#63 0x000055856b4399c1 in ?? ()
#64 0x000055856b43a207 in ?? ()
#65 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c14ef0) at /tmp/pear/temp/xdebug/src/base/base.c:779
#66 0x000055856b4399c1 in ?? ()
#67 0x000055856b43a207 in ?? ()
#68 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c14c60) at /tmp/pear/temp/xdebug/src/base/base.c:779
#69 0x000055856b4399c1 in ?? ()
#70 0x000055856b43a207 in ?? ()
#71 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c14b70) at /tmp/pear/temp/xdebug/src/base/base.c:779
#72 0x000055856b4399c1 in ?? ()
#73 0x000055856b43a207 in ?? ()
#74 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c14690) at /tmp/pear/temp/xdebug/src/base/base.c:779
#75 0x000055856b4399c1 in ?? ()
#76 0x000055856b43a207 in ?? ()
#77 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c14470) at /tmp/pear/temp/xdebug/src/base/base.c:779
#78 0x000055856b4399c1 in ?? ()
#79 0x000055856b43a207 in ?? ()
--Type <RET> for more, q to quit, c to continue without paging--
#80 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c14110) at /tmp/pear/temp/xdebug/src/base/base.c:779
#81 0x000055856b7665cc in ?? ()
#82 0x000055856b78bf93 in execute_ex ()
#83 0x00007f3f76bbc102 in xdebug_execute_ex (execute_data=0x7f3f76c14020) at /tmp/pear/temp/xdebug/src/base/base.c:779
#84 0x000055856b794ddf in zend_execute ()
#85 0x000055856b72868d in zend_execute_scripts ()
#86 0x000055856b6c4c31 in php_execute_script ()
#87 0x000055856b8088c1 in ?? ()
#88 0x000055856b444961 in ?? ()
#89 0x00007f3f7985fd0a in __scalbn (x=0, n=1844734143) at ../sysdeps/ieee754/dbl-64/wordsize-64/s_scalbn.c:35
#90 0x00007ffd34f820c8 in ?? ()
#91 0x0000000234f820c8 in ?? ()
#92 0x000055856b444590 in ?? ()
#93 0x00007f3f7a3c1180 in ?? ()
#94 0x0000000000000000 in ?? ()
(gdb)
```

**Comment by bwoebi** [comment]
Hey @gnumoksha,

Thanks for confirming that the error disappears with the upgrade.

Given that stacktrace, it's most likely that you stumbled over https://bugs.php.net/bug.php?id=81430, which was fixed in PHP 8.1.3.

**Comment by bwoebi** [comment]
Given that the error occurs on a reasonably often installed version (8.1.2 being the ubuntu default), we're going to implement a workaround.
